custom_components/tahoma/tahoma_device.py

```diff
@@ -27,43 +29,45 @@
 class TahomaDevice(Entity):
     """Representation of a TaHoma device entity."""
 
-    def __init__(self, tahoma_device, controller):
+    def __init__(self, device: Device, client: TahomaClient):
         """Initialize the device."""
-        self.tahoma_device = tahoma_device
-        self.controller = controller
+        self.device = device
+        self.client = client
         self._exec_queue = []
 
-    def update(self):
+    async def async_update(self):
         """Update method."""
-        if self.should_wait():
-            self.hass.add_job(self.update)
+        if await self.should_wait():
+            self.hass.async_add_job(self.async_update)
             return
-        self.controller.get_states([self.tahoma_device])
-        self.schedule_update_ha_state()
```

Comment:
This looks not necessary? Do we know why this has been added here in the first place?

Reply:
@vlebourl ?

Reply:
which part? last line? 
`schedule_update_ha_state()` writes down the states to HA. maybe not necessary in an update method indeed.

Reply:
I am not 100% sure, we should test this thoroughly.